### DISCUSSION TITLE
hyper 1.0 doesn't have `Body::wrap_stream`.  how do I use `UnboundedReceiverStream` as a Body ?

### DISCUSSION THREAD

**Discussion by Jhonfunk** [hyperium/hyper#3525] [Q&A]
```rust
// hyper = "0.14.27"

let response_stream = Body::wrap_stream(UnboundedReceiverStream::new(rx));

    Response::builder()
        .body(response_stream)
        .unwrap()
```

hyper 1.0 doesn't have `Body::wrap_stream`.  how do I use `UnboundedReceiverStream` as a Body ?

**Comment by seanmonstar** [comment]
You can wrap use the [StreamBody](https://docs.rs/http-body-util/latest/http_body_util/struct.StreamBody.html) adapter.

**Reply by Jhonfunk** [reply]
`StreamBody` seems to only accept `ReaderStream` and does not work with `UnboundedReceiverStream` or `ReceiverStream`.

**Reply by frederikhors** [reply]
Did you find a way?

**Reply by dswij** [reply]
[This](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=use+http_body_util%3A%3AStreamBody%3B%0Ause+hyper%3A%3AResponse%3B%0Ause+tokio_stream%3A%3Awrappers%3A%3AUnboundedReceiverStream%3B%0Ause+tokio%3A%3Async%3A%3Ampsc%3B%0A%0Afn+main%28%29+%7B%0A++++let+%28_tx%2C+rx%29+%3D+mpsc%3A%3Aunbounded_channel%3A%3A%3C%28%29%3E%28%29%3B%0A++++let+response_stream+%3D+StreamBody%3A%3Anew%28UnboundedReceiverStream%3A%3Anew%28rx%29%29%3B%0A++++let+_+%3D+Response%3A%3Abuilder%28%29%0A++++++++.body%28response_stream%29%0A++++++++.unwrap%28%29%3B%0A%7D) works.
